src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -152,38 +152,38 @@ void quietlyCloseConnection(final Connection connection, final String closureRea
    boolean isConnectionAlive(final Connection connection)
    {
       try {
-         try {
-            setNetworkTimeout(connection, validationTimeout);
-
-            final long validationSeconds = (int) Math.max(1000L, validationTimeout) / 1000;
+         setNetworkTimeout(connection, validationTimeout);
 
-            if (isUseJdbc4Validation) {
-               return connection.isValid((int) validationSeconds);
-            }
+         final int validationSeconds = (int) Math.max(1000L, validationTimeout) / 1000;
 
+         boolean valid = true;
+         if (isUseJdbc4Validation) {
+            valid = connection.isValid(validationSeconds);
+         }
+         else {
             try (Statement statement = connection.createStatement()) {
-               if (isNetworkTimeoutSupported != TRUE) {
-                  setQueryTimeout(statement, (int) validationSeconds);
+               if (isNetworkTimeoutSupported == FALSE) {
+                  setQueryTimeout(statement, validationSeconds);
                }
 
                statement.execute(config.getConnectionTestQuery());
-            }
-         }
-         finally {
-            setNetworkTimeout(connection, networkTimeout);
+             }
+          }
 
-            if (isIsolateInternalQueries && !isAutoCommit) {
-               connection.rollback();
-            }
-         }
+          if (valid) {
```

Comment:
And extra conditional.

Reply:
this is to avoid rollback and resetting networktimeout for dead / invalid connection